### ISSUE TITLE
Inconsistent wavenumber units and power spectrum normalisations

### ISSUE THREAD

**Issue by rrjbca** [skypyproject/skypy#442]
**Describe the bug**
The SkyPy implementation of the Eisenstein-Hu linear power spectrum is inconsistent with the wrappers around CAMB, CLASS in terms of the wavelength units and amplitude scaling.

**To Reproduce**

```python
>>> from astropy.cosmology import Planck18
>>> from matplotlib import pyplot as plt
>>> import numpy as np
>>> from skypy.power_spectrum import camb, classy, eisenstein_hu, growth_function

>>> k = np.logspace(-3, -1, 50)
>>> z, A_s, n_s = 0, 2.2E-9, 0.97

>>> p_camb = camb(k, z, Planck18, A_s, n_s)
>>> p_classy = classy(k, z, Planck18, A_s=A_s, n_s=n_s)
>>> p_eh = eisenstein_hu(k, A_s, n_s, Planck18)

>>> plt.plot(k, p_camb, label='CAMB')
>>> plt.plot(k, p_classy, label='CLASS')
>>> plt.plot(k, p_eh, label='Eisenstein-Hu')
>>> plt.loglog()
>>> plt.legend()
>>> plt.show()
```

![Figure_1](https://user-images.githubusercontent.com/58728519/109662840-64ea7d80-7b63-11eb-8f22-8a41d1ac6bcf.png)

Here it is clear that the peak of the power spectrum is shifted horizontally, indicating that the different models are making different assumptions about the units of the wavenumber k. Also there is an overall amplitude offset.

```python
>>> d = growth_function(z, Planck18)

>>> plt.plot(k, p_camb, label='CAMB')
>>> plt.plot(k, p_classy, label='CLASS')
>>> plt.plot(k*Planck18.h, p_eh/(d*d), label='Eisenstein-Hu')
>>> plt.loglog()
>>> plt.legend()
>>> plt.show()
```

![Figure_2](https://user-images.githubusercontent.com/58728519/109662854-69af3180-7b63-11eb-8ea4-8332efc0a8bb.png)

The second figure makes it clear that CAMB, CLASS and EH are not consistently assuming units of wavenumber to be Mpc (instead of Mpc/h; see [ADR-2](https://github.com/skypyproject/skypy/blob/main/docs/arch/adr-02.md)). Also the discrepancy in the normalisation could be a factor of the growth function at redshift zero squared although correcting for this does not give precise agreement.

**Expected behavior**
All power spectrum models should use the same wavenumber units and amplitude scaling conventions for consistency. I would suggest that modifying our implementation of EH to be consistent with the conventions used in CAMB/CLASS would make it more intuitive for users familiar with those codes, although I note that might go against our [ADR-2](https://github.com/skypyproject/skypy/blob/main/docs/arch/adr-02.md). Also (the option to) normalise based upon sigma_8 might be advantageous?

**Desktop (please complete the following information):**
 - SkyPy `module/halos` branch

**Comment by jucordero** [comment]
I'll take a look a this.

**Comment by rrjbca** [comment]
Thanks @jucordero. Discussing with @ntessore and @Lucia-Fonseca yesterday they pointed out that the amplitude scaling in my plots could be missing factors of `Planck18.h` (instead of `d`) but we could not converge on the solution.

**Comment by Lucia-Fonseca** [comment]
Growth functions by Linder and Carroll
![image](https://user-images.githubusercontent.com/55983939/109828499-95e6b300-7c34-11eb-9a50-e4e4bf4323d5.png)

The Carroll's growth function is the one needed for the Eisenstein & Hu power spectrum (see here after correcting for the `h` units)

```
pc = camb(wavenumber, redshift, Planck15, A_s, n_s)
peh = eisenstein_hu(wavenumber, A_s, n_s, Planck15, kwmap=0.02, wiggle=True)
gfc = growth_function_carroll(0, Planck15)

plt.loglog(wavenumber, pc[0], 'k', label='CAMB')
plt.loglog(wavenumber*h0, gfc**2*peh/h0**3,'--', label='EH', lw=2)
```

![image](https://user-images.githubusercontent.com/55983939/109828967-08579300-7c35-11eb-8013-c373b0a959ba.png)

**Comment by rrjbca** [comment]
Resolved by #445
